### PR TITLE
Add at-cfunction

### LINKED ISSUE #535: cfunction deprecation
https://github.com/JuliaLang/julia/pull/26486

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #535

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1707,4 +1707,10 @@ end
 @test isletter('β')
 @test !isletter('3')
 
+# 0.7.0-DEV.4762
+let ptr = @cfunction(+, Int, (Int, Int))
+    @test ptr isa Ptr{Cvoid}
+    @test ptr != C_NULL
```
[the remark this answers is not in the document]
Comment (ararslan) [reply test/runtests.jl]: Added

**Comment by martinholters** [comment]
AV failure due to #556. (Linux Travis seems to be using a Julia version old enough to not be hitting that.) OSX looks interesting:
```
 Resolving package versions...
ERROR: SystemError: unable to read directory /Users/travis/build/JuliaLang/Compat.jl/juliamnt/.Trashes: Permission denied
```
Pretty surely unrelated, but what's going on there?

**Comment by ararslan** [comment]
Yeah that's pretty bizarre, absolutely no idea.

**Comment by martinholters** [comment]
I've seen that elsewhere, too, but can't remember where just now. CC @KristofferC, @StefanKarpinski as it seems to be happing to package resolution or loading?

**Comment by KristofferC** [comment]
It is doing a `walkdir` in the packages path to find usages of standard libraries for old packages without project files. Perhaps we should use the `onerror` keyword to `walkdir` to make it more robust...

**Comment by martinholters** [comment]
Is there something we can do here (except putting Project.toml back)?

**Comment by KristofferC** [comment]
Nightlies are updated now so perhaps no harm putting the Project file back? That was the plan IIRC.

**Comment by KristofferC** [comment]
https://github.com/JuliaLang/Pkg.jl/pull/316 should fix it though.

**Comment by martinholters** [comment]
I'll close/re-open to re-trigger CI on top of the latest fixes and if that doesn't bring up anything except the known OSX failure, I'll go ahead and merge this later today.

**Comment by ararslan** [comment]
Thanks all!

**Pull request merged**